packages/components/src/Button/Button.test.tsx

```diff
@@ -166,15 +166,33 @@ it("routes when buttons are clicked", () => {
   expect(queryByText("Dos")).not.toBeInstanceOf(HTMLElement);
   expect(queryByText("Tres")).not.toBeInstanceOf(HTMLElement);
 
-  fireEvent.click(queryByText("Two"));
+  fireEvent.click(getByText("Two"));
 
   expect(queryByText("Uno")).not.toBeInstanceOf(HTMLElement);
   expect(queryByText("Dos")).toBeInstanceOf(HTMLElement);
   expect(queryByText("Tres")).not.toBeInstanceOf(HTMLElement);
 
-  fireEvent.click(queryByText("Three"));
+  fireEvent.click(getByText("Three"));
 
   expect(queryByText("Uno")).not.toBeInstanceOf(HTMLElement);
   expect(queryByText("Dos")).not.toBeInstanceOf(HTMLElement);
   expect(queryByText("Tres")).toBeInstanceOf(HTMLElement);
 });
+
+describe("Button role", () => {
+  it("should have a role of button when role not provided", () => {
+    const { getByRole } = render(<Button label="hello" />);
+    expect(getByRole("button")).toBeInstanceOf(HTMLButtonElement);
+  });
+  it("should not have a role of button when role not provided and it's a link", () => {
+    const { queryByRole, getByRole } = render(
+      <Button label="hello" url="myspace.com" />,
+    );
+    expect(queryByRole("button")).not.toBeInTheDocument();
```

Comment:
What is the reason for using queryByRole here and getByRole above?

Reply:
hey! yeah so `getBy*` and `queryBy*` each have their own use cases they are best suited for

>getBy...: Returns the matching node for a query, and throw a descriptive error if no elements match or if more than one match is found (use getAllBy instead if more than one element is expected).

so if you **are expecting something** to be there, you'd reach for `getBy*`

VS

>queryBy...: Returns the matching node for a query, and return null if no elements match. This is useful for asserting an element that is not present. Throws an error if more than one match is found (use queryAllBy instead if this is OK).

https://testing-library.com/docs/queries/about

so for this particular test I want to make sure this instance of `Button` that turns into an `<a>` doesn't have the `button` role AKA I want to search for something I expect to not exist - so `queryBy`

then the second assertion I'm making I do want it, and expect it, to exist so I use `getBy`

-------

as for my actual assertion logic you could definitely say that I'm not even passing in a `role="button"` to this invocation, and if I did - the test would in fact fail because it does end up adding the role to the output HTML

![image](https://github.com/GetJobber/atlantis/assets/11843215/f4984920-2750-49ae-947d-370edd25b781)

so arguably my test should/could be "Typescript should get upset if given a role for an incompatible button type" but that's testing our implementation of Typescript and the XOR package. we've done nothing to stop you ignoring Typescript and putting a `role` on a `Button` it doesn't belong on

with a quick search, it seems like it is possible to write tests for Typescript types, but we'd need to add a few packages & setup
https://www.olivare.net/blog/testing-ts-type-defs